From 61f7ab4e115bb2b843a151744df6f950b7bd2e5e Mon Sep 17 00:00:00 2001
From: Alexander Pipelka <alexander.pipelka@gmail.com>
Date: Wed, 27 Sep 2017 19:16:12 +0200
Subject: [PATCH] Save some ms on changing the tuner state.

---
 tuner.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tuner.c b/tuner.c
index 0cc0ad9..2cd2278 100644
--- a/tuner.c
+++ b/tuner.c
@@ -622,6 +622,7 @@ bool cSatipTuner::RequestState(eTunerState stateP, eStateMode modeP)
   else
      return false;
 
+  sleepM.Signal();
   return true;
 }
 
-- 
2.7.4

